### DISCUSSION TITLE
v1.0.0 Migration

### DISCUSSION THREAD

**Discussion by martijnvdbrug** [vendurehq/vendure#768] [General]
Post all your v1.0.0 related migration questions and finds here!

**Comment by martijnvdbrug** [comment]
I have found that running `migration:generate` generates a lot of unnecessary  `"ALTER TABLE X CHANGE 'updatedAt'...` lines. It looks like this is a TypeORM bug https://github.com/typeorm/typeorm/issues/3991 

I have verified that my `updatedAt` columns are already `datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6)`, so I have removed all migration lines that altered `updatedAt` columns.

I am not sure if it can do any harm if you decide to run these lines even though they don't change the columns?

**Comment by michaelbromley** [comment]
If you get the error `TypeError: record.factory is not a function` when building the Admin UI using `@vendure/ui-devkit`, see https://github.com/vendure-ecommerce/vendure/issues/758 for a work-around.

**Comment by martijnvdbrug** [comment]
If you have multiple channel's, don't forget to add `Assets` to your other channels. The migration script only adds them to the default channel.

My approach: 
1. Modify `addToDefaultChannel()` in `migration-utils.ts` to add all assets to all channels
2. Use the Admin UI to manually remove assets from the current channel, this will keep the assets in other channels

**Reply by martijnvdbrug** [reply]
The same goes for `PaymentMethod`, which was only added to the default channel. In my case the Handler arguments was a list, which did not migrate well.

I had to change my payment integration to work with 1.0.0, so migrating manually was not a big deal.

**Reply by martijnvdbrug** [reply]
And don't forget to add your `facets` and `facetValues` to any additional channels
